### DISCUSSION TITLE
Custom Axes scales beyond what's bundled

### DISCUSSION THREAD

**Discussion by phobson** [vega/vega-lite#8980] [Q&A]
Hey folks,

I've been getting deeper into vega-lite via Altair and I wasn't able to locate any examples or documentation about creating custom axes transformations and scales beyond the bundled [continuous scales](https://vega.github.io/vega-lite/docs/scale.html). 

To be specific, I wrote a library called [mpl-probscale](https://matplotlib.org/mpl-probscale/) that implements probability scales in (wait for it...) matplotlib. 

Short version: probability scales take a quantile plot and convert the quantiles to probabilities (and back) based on the CDF/PPF of some user-specified probability distribution (typically normal or weibull in my field). 

Here's an example comparing three different probability scales and a linear scale all with the same tick locations:
(here's a link to the [docs page](https://matplotlib.org/mpl-probscale/tutorial/getting_started.html#probabilty-scales), just scroll down a bit) 

![image](https://github.com/vega/vega-lite/assets/1163939/04db1d78-8f27-45c7-b151-4b4d7eea1609)

I'd love to be able to do something like this in vega-lite (and eventually Altair), but I'm getting a little lost in the internals. Can anyone point me in the right direction?

**Comment by declann** [comment]
Interesting!

I know that you can add custom scales in Vega - underlying Vega-Lite, here: https://vega.github.io/vega/docs/api/extensibility/#scales note the warning at the top though

Not sure if you will get to reference this scale type in Vega Lite or need to modify the Vega spec VL compiles into.

I guess the axis plotting will work as normal, pointing to a new scale in the spec

The Vega scale code is around here: https://github.com/vega/vega/blob/main/packages/vega-scale/src/scales/scaleBinOrdinal.js

Maybe nice feature request is for custom scales in Vega Lite and then Altair - since Vega itself appears to support it.
